connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -81,7 +81,26 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
         } else {
           child
         }
-      case _ => throw InvalidPlanInput()
+      case proto.Read.ReadTypeCase.FILE_DATA_SOURCE =>
+        val reader = session.read
+        if (rel.getFileDataSource.getFormat != null) {
+          reader.format(rel.getFileDataSource.getFormat)
+        }
+        if (rel.getFileDataSource.getSchema != null) {
```

Comment:
```suggestion
        if (rel.getFileDataSource.hasSchema) {
```

Reply:
This is the first I tried and I am pretty there is no `hasSchema`. This is so wried to me that the generate class does not have a `contains` like calls on these fields.